src/MiniProfiler.Shared/Timing.cs

```diff
@@ -253,19 +258,43 @@ public void Stop()
         /// </remarks>
         public void AddChild(Timing timing)
         {
-            if (Children == null)
+            if (Interlocked.Increment(ref _threadsAccessing) > 1)
+                throw new Exception("Concurrent access");
+
+            try
+            {
+                if (Children == null)
                 Children = new List<Timing>();
 
-            Children.Add(timing);
-            if(timing.Profiler == null)
-                timing.Profiler = Profiler;
-            timing.ParentTiming = this;
-            timing.ParentTimingId = Id;
-            if (Profiler != null)
-                timing.MiniProfilerId = Profiler.Id;
+                Children.Add(timing);
+                if(timing.Profiler == null)
+                    timing.Profiler = Profiler;
+                timing.ParentTiming = this;
+                timing.ParentTimingId = Id;
+                if (Profiler != null)
+                    timing.MiniProfilerId = Profiler.Id;
+            }
+            finally
+            {
+                Interlocked.Decrement(ref _threadsAccessing);
+            }
         }
 
-        internal void RemoveChild(Timing timing) => Children?.Remove(timing);
+        internal void RemoveChild(Timing timing)
+        {
+            if (Interlocked.Increment(ref _threadsAccessing) > 1)
+                throw new Exception("Concurrent access");
+
+            Task.Delay(10).Wait();
```

Comment:
i think this delay is helping to trigger the concurrency since the timing spend much longer doing RemoveChild